PZNS_Framework/media/lua/client/03_mod_core/PZNS_NPCSurvivor.lua

```diff
@@ -1,61 +1,204 @@
+        isForcedMoving = false,
+        isHoldingInPlace = false,
+        isMeleeOnly = false,
+        isRaider = false,
+        isSavedInWorld = false,
+        courage = 50,
+        jobName = "Guard",
+        jobSquare = nil,
+        isJobRefreshed = false,
+        currentAction = "",
+        followTargetID = "",
+        speechTable = nil,
+        lastEquippedMeleeWeapon = nil,
+        lastEquippedRangeWeapon = nil,
+        idleTicks = 0,
+        isStuckTicks = 0,
+        moveTicks = 0,
+        jobTicks = 0,
+        actionTicks = 0,
+        attackTicks = 0,
+        speechTicks = 0,
+        aimTarget = "",
+        canAttack = true,
+        canSaveData = true,
+        textObject = nil,
+        ----- Relations -----
+        relationsMap = {},
+        initialRelation = 0,
+        _minRelation = 0,
+        _maxRelation = 50,
+        anonymousMap = {},
+        affection = 50,
         ------ IsoPlayer Spawning Related below ------
-        isAlive = true,                         -- WIP - Technically part of IsoPlayer; used when PZNS needs to use it before IsoPlayer is loaded.
-        isSpawned = true,                       -- WIP - Technically part of IsoPlayer... but "isExistInTheWorld()" seems very inconsistent...
-        forename = "",                          -- Cows: Placeholder; technically part of IsoPlayer; used when PZNS needs to use it before IsoPlayer is loaded.
-        surname = "",                           -- Cows: Placeholder; technically part of IsoPlayer; used when PZNS needs to use it before IsoPlayer is loaded.
-        isFemale = false,                       -- Cows: Placeholder; technically part of IsoPlayer; used when PZNS needs to use it before IsoPlayer is loaded.
-        squareX = nil,                          -- Cows: Placeholder; technically part of IsoPlayer; used when PZNS needs to use it before IsoPlayer is loaded.
-        squareY = nil,                          -- Cows: Placeholder; technically part of IsoPlayer; used when PZNS needs to use it before IsoPlayer is loaded.
-        squareZ = nil,                          -- Cows: Placeholder; technically part of IsoPlayer; used when PZNS needs to use it before IsoPlayer is loaded.
-        npcIsoPlayerObject = npcIsoPlayerObject -- Cows: objects cannot be saved to moddata...
+        isAlive = true,
+        isSpawned = true,
+        forename = "",
+        surname = "",
+        isFemale = false,
+        squareX = nil,
+        squareY = nil,
+        squareZ = nil,
+        npcIsoPlayerObject = npcIsoPlayerObject
     };
 
+    setmetatable(npcSurvivor, self);
+    self.__index = self;
+
     return npcSurvivor;
 end
+
+---Assigns groupID to NPC
+---@param groupID groupID
+function NPC:setGroupID(groupID)
+    self.groupID = groupID
+end
+
+---Unassign groupID from NPC
+function NPC:unsetGroupID()
+    self.groupID = nil
+end
+
+---comments
+---@param mpPlayerID integer
+function NPC:setPlayerNum(mpPlayerID)
```

Comment:
I don't know if this is actually necessary. I have been told that playerNum is local and that multiplayer handles player IDs and numbers differently. 
This probably isn't needed but if it's harmless, I'll leave it be.

Reply:
This parameter was added because I wanted to treat human player(s) as NPC objects (for displaying in UI, handling relations and such). I've already added player initialization [here](https://github.com/lanceris/PZNS/blob/940c39d09b11555f4eb9ce729b388490f17a53be/PZNS_Framework/media/lua/client/02_mod_utils/PZNS_PlayerUtils.lua#L27), but accidently included this parameter when transferring changes from other branch.